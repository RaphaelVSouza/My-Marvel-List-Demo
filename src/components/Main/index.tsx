import Menu from "components/Menu";
import React from "react";

import * as S from "./styles";

const Main = ({ description = "CRA, TypeScript e Styled Components" }) => (
  <S.Wrapper>
    <S.Logo
      src={`${process.env.PUBLIC_URL}/logo.svg`}
      alt="Imagem de um átomo e React Avançado escrito ao lado."
    />
    <S.Description>{description}</S.Description>
    <S.Illustration
      src={`${process.env.PUBLIC_URL}/hero-illustration.svg`}
      alt="Um desenvolvedor de frente para uma tela com código."
    />

    <Menu />
  </S.Wrapper>
);

export default Main;
