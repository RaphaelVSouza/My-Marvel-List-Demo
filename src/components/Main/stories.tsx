import { Story, Meta } from "@storybook/react";
import Main from ".";

export default {
  title: "Main",
  component: Main,
  args: {
    description: "description default",
  },
} as Meta;

export const Basic: Story = (args) => <Main {...args} />;
Basic.args = {
  description: "description basic",
};

export const Default: Story = (args) => <Main {...args} />;
