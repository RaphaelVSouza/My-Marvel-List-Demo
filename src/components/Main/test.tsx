import { render, screen } from "@testing-library/react";

import Main from ".";

// When you have nested components
// you don't need to test the inner component
// since it already has its own tests
// so you mock instead of rendering everything from it
jest.mock("components/Menu", () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Menu"></div>;
    },
  };
});

describe("<Main />", () => {
  it("should render the heading", () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole("heading", {
        name: /CRA, TypeScript e Styled Components/i,
      })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render the menu", () => {
    render(<Main />);

    expect(screen.getByTestId("Menu")).toBeInTheDocument();
  });

  it("should render the colors correctly", () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toHaveStyle({ "background-color": "#06092b" });
  });
});
