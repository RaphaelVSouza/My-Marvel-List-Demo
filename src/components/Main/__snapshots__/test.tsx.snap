// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="sc-bdnxRM fBbHXb"
>
  <img
    alt="Imagem de um átomo e React Avançado escrito ao lado."
    class="sc-gtsrHT gNmutz"
    src="/logo.svg"
  />
  <h2
    class="sc-dlnjwi gVpAup"
  >
    CRA, TypeScript e Styled Components
  </h2>
  <img
    alt="Um desenvolvedor de frente para uma tela com código."
    class="sc-hKFxyN fWLaFx"
    src="/hero-illustration.svg"
  />
  <div
    data-testid="Menu"
  />
</main>
`;
